src/main/java/filter/lexer/Lexer.java

```diff
@@ -21,6 +21,7 @@
             // These have higher priority than Symbol
             new Rule("\\d{4}-\\d{1,2}-\\d{1,2}", TokenType.DATE), // YYYY-MM?-DD?
             new Rule("[A-Za-z]+(-[A-Za-z]+)*\\s*:", TokenType.QUALIFIER),
+            new Rule("[A-Za-z0-9#][/A-Za-z0-9.'+-]+#", TokenType.COMPOUND_ID),
```

Comment:
Do we want to enforce the pattern preceding before `#` to fits a repoId structure i.e `test/test` ? I think it is better to allow any valid symbol but I do agree that we can enforce the `id` after the `#`.